pkg/event/processor_test.go

```diff
@@ -449,3 +448,153 @@ func TestChanQueueEventProcessor_ProcessBatch(t *testing.T) {
+	}
+
+	for _, merge := range merges {
+		b.Run(merge.name, merge.fun)
+	}
+}
+
+func BenchmarkWithBatchSize(b *testing.B) {
+	logging.SetLogger(&NoOpLogger{})
+
+	merges := []struct {
+		name string
+		batchSize int
+		fun  func(bs int, b *testing.B) int
+	}{
+		{"BatchSize10", 10, benchmarkProcessorWithBatchSize},
+		{"BatchSize20", 20,benchmarkProcessorWithBatchSize},
+		{"BatchSize30", 30, benchmarkProcessorWithBatchSize},
+		{"BatchSize40", 40,benchmarkProcessorWithBatchSize},
+		{"BatchSize50", 50, benchmarkProcessorWithBatchSize},
+		{"BatchSize60", 60, benchmarkProcessorWithBatchSize},
+	}
+
+	for _, merge := range merges {
+		b.Run(merge.name, func (b *testing.B) {
+			benchmarkProcessorWithBatchSize(merge.batchSize, b)
+		})
+	}
+
+}
+
+func BenchmarkWithQueue(b *testing.B) {
+	logging.SetLogger(&NoOpLogger{})
+
+	b.Run("InMemoryQueue", func (b *testing.B) {
+		benchmarkProcessorQueue(NewInMemoryQueue(defaultQueueSize), b)
+	})
+
+	b.Run("ChannelQueue", func (b *testing.B) {
+		benchmarkProcessorQueue(NewChanQueue(defaultQueueSize), b)
+	})
+
+}
+
+func benchmarkProcessor100(b *testing.B) {
+	benchmarkProcessorQueueSize(100, b)
+}
+
+func benchmarkProcessor500(b *testing.B) {
+	benchmarkProcessorQueueSize(500, b)
+}
+
+func benchmarkProcessor1000(b *testing.B) {
+	benchmarkProcessorQueueSize(1000, b)
+}
+
+func benchmarkProcessor2000(b *testing.B) {
+	benchmarkProcessorQueueSize(2000, b)
+}
+
+func benchmarkProcessor3000(b *testing.B) {
+	benchmarkProcessorQueueSize(3000, b)
+}
+
+func benchmarkProcessor4000(b *testing.B) {
+	benchmarkProcessorQueueSize(4000, b)
+}
+
+func benchmarkProcessorQueueSize(qSize int, b *testing.B) int {
+	exeCtx := utils.NewCancelableExecutionCtx()
+	dispatcher := NewMockDispatcher(100, false)
+	processor := NewBatchEventProcessor(
+		WithQueueSize(qSize),
+		WithEventDispatcher(dispatcher))
+	processor.Start(exeCtx)
+
+	conversion := BuildTestConversionEvent()
+
+	for i := 0; i < b.N; i++ {
+		processor.ProcessEvent(conversion)
+	}
+
+	exeCtx.TerminateAndWait()
+
+	return dispatcher.Events.Size()
+}
+
+func benchmarkProcessorQueue(q Queue, b *testing.B) int {
+	exeCtx := utils.NewCancelableExecutionCtx()
+	dispatcher := NewMockDispatcher(100, false)
+	processor := NewBatchEventProcessor(
+		WithQueue(q),
+		WithEventDispatcher(dispatcher))
+	processor.Start(exeCtx)
+
+	conversion := BuildTestConversionEvent()
+
+	for i := 0; i < b.N; i++ {
+		processor.ProcessEvent(conversion)
+	}
+
+	exeCtx.TerminateAndWait()
+
+	return dispatcher.Events.Size()
+}
+
+func benchmarkProcessorWithBatchSize(bs int, b *testing.B) int {
+	exeCtx := utils.NewCancelableExecutionCtx()
+	dispatcher := NewMockDispatcher(100, false)
+	processor := NewBatchEventProcessor(
+		WithBatchSize(bs),
+		WithEventDispatcher(dispatcher))
+	processor.Start(exeCtx)
+
+	conversion := BuildTestConversionEvent()
+
+	for i := 0; i < b.N; i++ {
+		processor.ProcessEvent(conversion)
+	}
+
+	exeCtx.TerminateAndWait()
+
+	return dispatcher.Events.Size()
```

Comment:
Are these `return` values use for anything?

Reply:
yes, they are now used in the queue size tests to make sure that all events are processed.